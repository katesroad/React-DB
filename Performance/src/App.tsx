import * as React from 'react'
import { Switch, Route, Link } from 'react-router-dom'
import { BrowserRouter } from 'react-router-dom'
import { Spinner } from 'components/common/Spinner'
import { ErrorBoundary } from 'react-error-boundary'
import { ErrorFallback } from 'components/common/ErrorBoundary'

/**
 * Add prefetch setting for code that user will load later
 * Doc: https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules
 */
const HomeScreen = React.lazy(
  () => import(/* webpackPrefetch: true */ './screens/HomeScreen')
)
const AboutScreen = React.lazy(
  () => import(/* webpackPrefetch: true */ './screens/AboutScreen')
)

function App() {
  return (
    <BrowserRouter>
      <header>
        <Link to="/">home</Link>
        <Link to="/about">about</Link>
      </header>
      {/* 
        About:Route based code splitting
        Doc: https://reactjs.org/docs/code-splitting.html#route-based-code-splitting
     */}
      <ErrorBoundary FallbackComponent={ErrorFallback}>
        <React.Suspense fallback={<Spinner />}>
          <Switch>
            {/* Split code using React.lazy and React.Suspense */}
            <Route path="/" exact component={HomeScreen} />
            <Route path="/about" exact component={AboutScreen} />
          </Switch>
        </React.Suspense>
      </ErrorBoundary>
    </BrowserRouter>
  )
}

export default App
