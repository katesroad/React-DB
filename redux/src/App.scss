.App {
  .heading {
    text-align: center;
  }
  ul,
  li {
    list-style: none;
  }
}
