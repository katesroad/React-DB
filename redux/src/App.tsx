import React from 'react'
import Voter from 'components/Voter'
import { useDispatch, useSelector } from 'react-redux'
import { RootState } from 'store'
import { getUser } from 'store/features/user'

function App() {
  const voters = ['John', 'Julian', 'Kate']
  const votes = useSelector((state: RootState) => state?.counter?.count || 0)
  const user = useSelector((state: RootState) => state?.user?.user)
  const { isLoading, isSuccess } = useSelector((state: RootState) => state?.app)

  const dispatch = useDispatch()
  React.useEffect(() => {
    dispatch(getUser())
  }, [dispatch])

  return (
    <div>
      <div className="heading">
        <h2>
          Votes for
          {isLoading ? (
            <span>loading...</span>
          ) : isSuccess ? (
            <span>{' ' + user?.lastName}</span>
          ) : null}
        </h2>
        <p>
          <strong>Total votes:</strong> {votes} votes
        </p>
      </div>
      <ul>
        {voters.map((name) => (
          <li key={name}>
            <Voter name={name} />
          </li>
        ))}
      </ul>
    </div>
  )
}

export default App
