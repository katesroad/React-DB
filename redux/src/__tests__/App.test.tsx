import React from 'react'
import { render, screen } from '@testing-library/react'
import App from '../App'
import AppProviders from 'context'

test('Render <App />', async () => {
  const { container } = render(<App />, { wrapper: AppProviders })

  expect(screen.getByText(/loading/i)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
