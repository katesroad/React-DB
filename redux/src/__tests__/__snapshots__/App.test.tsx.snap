// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <App /> 1`] = `
<div>
  <div>
    <div
      class="heading"
    >
      <h2>
        Votes for
        <span>
          loading...
        </span>
      </h2>
      <p>
        <strong>
          Total votes:
        </strong>
         
        0
         votes
      </p>
    </div>
    <ul>
      <li>
        <div
          class="sc-bdnxRM edcOPF"
        >
          <h2>
             
            John
             
          </h2>
          <h3>
             
            Votes: 0
             
          </h3>
          <div>
            <button
              class="sc-gtsrHT jPIbuq"
            >
               Increment 
            </button>
            <button
              class="sc-gtsrHT jPIbuq"
              disabled=""
            >
              Decrement
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-bdnxRM edcOPF"
        >
          <h2>
             
            Julian
             
          </h2>
          <h3>
             
            Votes: 0
             
          </h3>
          <div>
            <button
              class="sc-gtsrHT jPIbuq"
            >
               Increment 
            </button>
            <button
              class="sc-gtsrHT jPIbuq"
              disabled=""
            >
              Decrement
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-bdnxRM edcOPF"
        >
          <h2>
             
            Kate
             
          </h2>
          <h3>
             
            Votes: 0
             
          </h3>
          <div>
            <button
              class="sc-gtsrHT jPIbuq"
            >
               Increment 
            </button>
            <button
              class="sc-gtsrHT jPIbuq"
              disabled=""
            >
              Decrement
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
